pkg/providers/vmware/client/client.go

```diff
@@ -0,0 +1,131 @@
+package client
+
+import (
+	"context"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+	"net/url"
+	"time"
+)
+
+// timeout value in seconds for vmware api requests
+const timeout = 5 * time.Second
```

Comment:
Is this timeout OK for any vmware request, or can some request take more time? I just wonder whether this should be somehow configurable for some slow envs.

Reply:
I'm not sure; I was hoping to get some feedback from the team on that.

Reply:
Some of the envs could be slower. Are we going to retry if a request would time out?

Reply:
I was thinking the context error that gets propagated up to Reconcile would result in a requeue (so it wouldn't block the reconcile for other resources, but would still retry via a requeue), but I may be mistaken about that.